cirq/ops/global_phase_op_test.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest
+
+import cirq
+
+
+def test_init():
+    op = cirq.GlobalPhaseOperation(1j)
+    assert op.coefficient == 1j
+    assert op.qubits == ()
+    assert op.with_qubits() is op
+
+    with pytest.raises(ValueError, match='not unitary'):
+        _ = cirq.GlobalPhaseOperation(2)
+    with pytest.raises(ValueError, match='0 qubits'):
+        _ = cirq.GlobalPhaseOperation(1j).with_qubits(cirq.LineQubit(0))
+
+
+def test_protocols():
+    for p in [1, 1j, -1]:
+        cirq.testing.assert_implements_consistent_protocols(
+            cirq.GlobalPhaseOperation(p))
+
+    np.testing.assert_allclose(cirq.unitary(cirq.GlobalPhaseOperation(1j)),
+                               np.array([[1j]]),
+                               atol=1e-8)
+
+
+def test_str():
+    assert str(cirq.GlobalPhaseOperation(1j)) == '1j'
+
+
+def test_diagram():
+    a, b = cirq.LineQubit.range(2)
+    x, y = cirq.LineQubit.range(10, 12)
+
+    cirq.testing.assert_has_diagram(
+        cirq.Circuit([
+            cirq.Moment([
+                cirq.CNOT(a, x),
+                cirq.CNOT(b, y),
+                cirq.GlobalPhaseOperation(-1),
+            ])
+        ]), """
+                ┌──┐
+0: ──────────────@─────
+                 │
+1: ──────────────┼@────
+                 ││
+10: ─────────────X┼────
+                  │
+11: ──────────────X────
+
+global phase:    π
+                └──┘
+        """)
+
+    cirq.testing.assert_has_diagram(
+        cirq.Circuit([
+            cirq.Moment([
+                cirq.CNOT(a, x),
+                cirq.CNOT(b, y),
+                cirq.GlobalPhaseOperation(-1),
+                cirq.GlobalPhaseOperation(-1),
+            ]),
+        ]), """
+                ┌──┐
+0: ──────────────@─────
+                 │
+1: ──────────────┼@────
+                 ││
+10: ─────────────X┼────
+                  │
+11: ──────────────X────
+
+global phase:
```

Comment:
this empty line seems a little off. I understand why though: there are `GlobalPhaseOperation` so you need to have a line for `global phase` but you don't want to print a `1` for every moment that doesn't have a non-unity global phase